pkg/adsc/adsc.go

```diff
@@ -383,7 +498,18 @@ func (a *ADSC) handleRecv() {
 					_ = proto.Unmarshal(valBytes, el)
 					eds = append(eds, el)
 				}
-
+			case v2.EndpointType:
```

Comment:
It's an XDS client - maybe it'll talk with a different XDS server that is not istiod/pilot.

Reply:
I highly suspect this will not actually work, as you are trying to marshal a v2 type into a v3 struct. You need to modify the typeurl manually for it to work